File: python/stdlib/structures/if.py
try:
    from ...stdlib import Node as base
except ValueError:
    from stdlib import Node as base


class Node(base.Node):
    def __init__(self, verbose, args):
        super(Node, self).__init__("If", "stdlib.structures.if", "",
                                   {"condition":"Boolean", "val":"Object"},
                                   {"true": "Object", "false":"Object"},
                                   "If condition pass to true otherwise to false.", verbose)

    def tick(self, value):
        if value["condition"]:
            return {"true": value["val"], "false": None}
        else:
            return {"true": None, "false": value["val"]}

File: lua/stdlib/structures/if.lua
local myNode = {}
function myNode.init(node)
  node.tick = function(value)
    result = {}
    result["true"] = nil
    result["false"] = nil
    if value.condition then
      result["true"] = value.val
    else
      result["false"] = value.val
    end
    return result
  end
end

function myNode.spec(node)
  node.name = "If"
  node.inputs.val = "Object"
  node.inputs.condition = "Boolean"
  node.outputs["true"] = "Object"
  node.outputs["false"] = "Object"
  node.desc = "Pass val based on condition."
end

return myNode
